### PR TITLE
feat(NGMA-1113): add getCategoryById query

### PR DESCRIPTION
**Pull request by Willbrighte**
Added getCategoryById to finance core api.

### CODE REVIEW

**Review by fahadbrighte — DISMISSED** [review]

File: src/FinanceCoreApi.php [hunk 1]
```diff
@@ -338,4 +339,64 @@ private function getFinanceAccountFromResponse($data): Account
...
+    /**
+     * getCategoryById
+     *
+     * @param  int $categoryId
+     * @return Category|null
+     */
+    public function getCategoryById(int $categoryId): ?Category
+    {
+        $query = <<<GQL
+            query {
```
Comment (christheyounger) [inline src/FinanceCoreApi.php:352]: suggestion: add a name for this query and input definitions as per convention
```
query GetCategory($categoryId: Int!) {
````

File: src/FinanceCoreApi.php [hunk 2]
```diff
@@ -338,4 +339,64 @@ private function getFinanceAccountFromResponse($data): Account
...
+                category(id: $categoryId) {
+                    id
+                    slug
+                    name
+                    icon
+                    isVirtualPowerPlantCompatible
```
Comment (christheyounger) [inline src/FinanceCoreApi.php:358]: question: you've requested a lot of properties that you don't set on the output of the function. should we remove them?

**Review by christheyounger — CHANGES_REQUESTED** [review]
Looking good, but I ask please for a couple of small changes. thanks

**Review by christheyounger — CHANGES_REQUESTED** [review]
Looking good, but I ask please for a couple of small changes. thanks

File: src/FinanceCoreApi.php (hunk 2 above)
Comment (Willbrighte) [reply src/FinanceCoreApi.php:358]: Will remove them, thanks Chris.

File: src/FinanceCoreApi.php (hunk 1 above)
Comment (Willbrighte) [reply src/FinanceCoreApi.php:352]: Will update.

**Review by fahadbrighte — APPROVED** [review]

**Review by christheyounger — APPROVED** [review]

**Pull request merged**